pystiche/papers/ulyanov_et_al_2016/data.py

```diff
@@ -0,0 +1,269 @@
+        ),
+        "pebble": DownloadableImage(
+            urljoin(base_ulyanov_suppl_texture, "pebble.png"),
+            md5="5b5e5aa6c579e42e268058a94d683a6c",
+        ),
+        "pixelcity_windows2": DownloadableImage(
+            urljoin(base_ulyanov_suppl_texture, "pixelcity_windows2.jpg"),
+            md5="53026a8411e7c26e959e36d3223f3b8f",
+        ),
+        "red-peppers256.o": DownloadableImage(
+            urljoin(base_ulyanov_suppl_texture, "red-peppers256.o.jpg"),
+            md5="16371574a10e0d10b88b807204c4f546",
+        ),
+    }
+    base_johnson = (
+        "https://raw.githubusercontent.com/jcjohnson/fast-neural-style/master/images/"
+    )
+    style_base_johnson = urljoin(base_johnson, "styles/")
+
+    base_ulyanov_suppl_style = "https://raw.githubusercontent.com/DmitryUlyanov/texture_nets/texture_nets_v1/supplementary//stylization_models/"
+    style_images = {
+        "candy": DownloadableImage(
+            urljoin(style_base_johnson, "candy.jpg"),
+            md5="00a0e3aa9775546f98abf6417e3cb478",
+        ),
+        "the_scream": DownloadableImage(
+            urljoin(style_base_johnson, "the_scream.jpg"),
+            md5="619b4f42c84d2b62d3518fb20fa619c2",
+        ),
+        "Jean Metzinger": DownloadableImage(
+            "https://upload.wikimedia.org/wikipedia/commons/c/c9/Robert_Delaunay%2C_1906%2C_Portrait_de_Metzinger%2C_oil_on_canvas%2C_55_x_43_cm%2C_DSC08255.jpg",
+            title="Portrait of Jean Metzinger",
+            author="Jean Metzinger",
+            date="1906",
+            license=PublicDomainLicense(1906),
+            md5="3539d50d2808b8eec5b05f892d8cf1e1",
+            file="jean_metzinger.jpg",
+        ),
+        "mosaic": DownloadableImage(
+            urljoin(base_ulyanov_suppl_style, "mosaic.jpg"),
+            md5="4f05f1e12961cebf41bd372d909342b3",
+        ),
+        "pleades": DownloadableImage(
+            urljoin(base_ulyanov_suppl_style, "pleades.jpg"),
+            md5="6fc41ac30c2852a5454a0ead2f479dc9",
+        ),
+        "starry": DownloadableImage(
+            urljoin(base_ulyanov_suppl_style, "starry.jpg"),
+            md5="c6d94f7962466b2e80a64ae82523242a",
+        ),
+        "turner": DownloadableImage(
+            urljoin(base_ulyanov_suppl_style, "turner.jpg"),
+            md5="7fdd9603a5182dcef23d7fb1c5217888",
+        ),
+    }
+    return DownloadableImageCollection(
+        {**texture_images, **content_images, **style_images},
+        root=root,
+        download=download,
+        overwrite=overwrite,
+    )
+
+
+def ulyanov_et_al_2016_dataset(
+    root: str,
+    impl_params: bool = True,
+    instance_norm: bool = True,
+    transform: Optional[Transform] = None,
+):
+    if transform is None:
+        transform = ulyanov_et_al_2016_content_transform(
+            impl_params=impl_params, instance_norm=instance_norm
+        )
+    return ImageFolderDataset(root, transform=transform)
```

Comment:
Couldn't we use `ImageNet` from `torchvision` here? If yes, I will explain how.

Reply:
As discussed above, `ImageNet` should be used here. So yes. 
The following things should be possible: 
1. Selection between training data, validation data or both (depending on the parameters `instance_norm` and `impl_params`).
2. A random or fixed sequence of data. 
3. Only images with a certain image size (`height > 256` and `width > 256`)

Reply:
I'll take that up. Until then, just continue with what we have now.